test/common_utils.py

```diff
@@ -392,34 +392,38 @@ def assertEqual(self, x, y, prec=None, message='', allow_inf=False):
             self.assertEqual(x, y.item(), prec, message, allow_inf)
         elif isinstance(x, torch.Tensor) and isinstance(y, torch.Tensor):
             def assertTensorsEqual(a, b):
-                super(TestCase, self).assertEqual(a.size(), b.size(), message)
-                if a.numel() > 0:
-                    if a.device.type == 'cpu' and a.dtype == torch.float16:
-                        # CPU half tensors don't have the methods we need below
-                        a = a.to(torch.float32)
-                    if TEST_WITH_ROCM:
-                        # Workaround for bug https://github.com/pytorch/pytorch/issues/16448
-                        # TODO: remove after the bug is resolved.
-                        b = b.to(a.dtype).to(a.device)
-                    else:
-                        b = b.to(a)
-                    diff = a - b
-                    if a.is_floating_point():
-                        # check that NaNs are in the same locations
-                        nan_mask = torch.isnan(a)
-                        self.assertTrue(torch.equal(nan_mask, torch.isnan(b)), message)
-                        diff[nan_mask] = 0
-                        # inf check if allow_inf=True
-                        if allow_inf:
-                            inf_mask = torch.isinf(a)
-                            inf_sign = inf_mask.sign()
-                            self.assertTrue(torch.equal(inf_sign, torch.isinf(b).sign()), message)
-                            diff[inf_mask] = 0
-                    # TODO: implement abs on CharTensor (int8)
-                    if diff.is_signed() and diff.dtype != torch.int8:
-                        diff = diff.abs()
-                    max_err = diff.max()
-                    self.assertLessEqual(max_err, prec, message)
+                if isinstance(x, torch.BoolTensor) and isinstance(y, torch.BoolTensor):
```

Comment:
`isinstance` is the legacy way to test the dtype of a tensor. The modern way is to check if `x.dtype == torch.bool` (or whatever you've called the dtype).

Also, I'd suggest adjusting where exactly you place the conditional here. Notice in the code you short-circuited out, we do a size check first. This is generally useful and you should keep that test even in the boolean case. It seems to that you should put the conditional around `diff = a - b` and subsequent lines.

Reply:
Also, defensive programming says that you should explicitly check for the case when only one or the other tensor is bool, and raise an error in that  case (since it means programmer error), rather than let it fall to subtraction and get some sort of weird error that you tried to subtract a float from a bool, e.g.